drivers/storage/portworx/components_test.go

```diff
@@ -50,11 +52,41 @@ import (
 	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
 	"github.com/libopenstorage/operator/pkg/constants"
 	"github.com/libopenstorage/operator/pkg/mock"
+	"github.com/libopenstorage/operator/pkg/mock/mockcore"
 	"github.com/libopenstorage/operator/pkg/util"
 	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
 	testutil "github.com/libopenstorage/operator/pkg/util/test"
 )
 
+// For some reason simpleClientset doesn't work with createToken, erroring out with serviceaccounts "" not found.
+// Thus wrap the simpleClientset with MockCoreOps.
+func setUpMockCoreOps(mockCtrl *gomock.Controller, clientset *fakek8sclient.Clientset) *mockcore.MockOps {
+	mockCoreOps := mockcore.NewMockOps(mockCtrl)
+	coreops.SetInstance(mockCoreOps)
+
+	simpleClientset := coreops.New(clientset)
+	mockCoreOps.EXPECT().
+		CreateToken(gomock.Any(), gomock.Any(), gomock.Any()).
+		Return(&authv1.TokenRequest{
+			Status: authv1.TokenRequestStatus{
+				Token: "",
+			},
+		}, nil).
+		AnyTimes()
+	mockCoreOps.EXPECT().
+		GetVersion().
+		Return(clientset.Discovery().ServerVersion()).
+		AnyTimes()
+	mockCoreOps.EXPECT().
+		ResourceExists(gomock.Any()).
+		Return(simpleClientset.ResourceExists(schema.GroupVersionKind{
```

Comment:
This is fine. But just curious. If this test file was calling more k8s functions, we would have to write an EXPECT() for each of them to redirect that call to simpleClientSet, right?

Reply:
Yes. This is done in the file `status_test.go`